core/domain/classifier_services.py

```diff
@@ -124,6 +124,57 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def train(exploration):
```

Comment:
@seanlip . 
(i) There is no urgency to merge this PR right now. This is just a feature we'd need to implement later anyway with slightly modified models. I can't stress about why this is important right now. @anmolshkl thought that this feature might be needed since we will be reusing this function along with the conditions. 
(ii) This PR will change a lot of things about how classifiers are trained. The ClassifyHandler hasn't been modified, so training will still happen from the learner's view. But, with this PR, the training will also happen whenever the classifier conditions are True and whenever _save_Exploration() or _create_exploration() is triggered. So,this moves the training to the creator's view rather than the learners view. If this is merged, creating/ updating an exploration would take more time than usual because the training also happens there. 
We'll have to weigh in the effects. I'll leave the decision of what to do with this PR to you and @anmolshkl